contracts/nft/src/execute/mint.rs

```diff
@@ -0,0 +1,167 @@
+    ResultStatus, Scope, Session,
+};
+use sha2::{Digest, Sha256};
+use uuid::Uuid;
+
+use crate::storage::nft::{Nft, TOKENS};
+use crate::storage::nft_count;
+use crate::util::metadata_address::MetadataAddress;
+use crate::{
+    core::error::ContractError,
+    storage,
+    util::action::{Action, ActionType},
+};
+
+pub fn handle(
+    deps: DepsMut,
+    _info: MessageInfo,
+    env: Env,
+    scope_uuid: Uuid,
+    session_uuid: Uuid,
+    recipient: Addr,
+) -> Result<Response, ContractError> {
+    let state = storage::state::get(deps.storage)?;
+    let contract_spec_uuid = Uuid::from_str(&state.contract_spec_uuid).unwrap();
+    let scope_spec_uuid = Uuid::from_str(&state.scope_spec_uuid).unwrap();
+
+    let scope = Scope {
+        scope_id: vec![],
+        specification_id: MetadataAddress::scope_specification(scope_spec_uuid)
+            .unwrap()
+            .bytes,
+        owners: vec![Party {
+            address: env.contract.address.to_string(),
+            role: PartyType::Provenance.into(),
+            optional: false,
+        }],
+        data_access: vec![],
+        value_owner_address: recipient.to_string(),
+        require_party_rollup: false,
+    };
+
+    let write_scope_msg = MsgWriteScopeRequest {
+        scope: Some(scope),
+        signers: vec![env.contract.address.to_string(), recipient.to_string()],
+        scope_uuid: scope_uuid.to_string(),
+        spec_uuid: scope_spec_uuid.to_string(),
+    };
+
+    let session = Session {
+        session_id: MetadataAddress::session(scope_uuid, session_uuid)
+            .unwrap()
+            .bytes,
+        specification_id: MetadataAddress::contract_specification(contract_spec_uuid)
+            .unwrap()
+            .bytes,
+        parties: vec![Party {
+            address: env.contract.address.to_string(),
+            role: PartyType::Provenance.into(),
+            optional: false,
+        }],
+        name: "nft_session".to_string(),
+        context: vec![],
+        audit: None,
+    };
+
+    let write_session_msg = MsgWriteSessionRequest {
+        session: Some(session.clone()),
+        signers: vec![env.contract.address.to_string()],
+        session_id_components: None,
+        spec_uuid: contract_spec_uuid.to_string(),
+    };
+
+    let record_specification = RecordSpecification {
+        specification_id: MetadataAddress::record_specification(
+            contract_spec_uuid,
+            "nft_owner_record_spec".to_string(),
+        )?
+        .bytes,
+        name: "nft_owner_record_spec".to_string(),
+        inputs: vec![InputSpecification {
+            name: "owner".to_string(),
+            type_name: "cosmwasm_std::Addr".to_string(),
+            // hash of cosmwasm_std v1.3.1 (from Cargo.lock)
+            source: Some(input_specification::Source::Hash(
+                "eb5e05a95fd2a420cca50f4e94eb7e70648dac64db45e90403997ebefeb143bd".to_string(),
```

Comment:
Why is this hash used?

Reply:
This is part of how the records work. it needs a way to verify what form/source the data will be in (if i understand it all correctly). so since it stores the `Addr` of the owner, it'd be that of cosmwasm_std... thoughts?